File: best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if(prices.size() == 0)
            return 0;
        int min_running = prices[0];
        int max_ = 0;
        for(int i = 0; i < prices.size(); ++i ) {
            if (prices[i] < min_running) {
                min_running = prices[i];
                int max_ = 0;
                
            }
            max_ = max(prices[i] - min_running, max_);
            
        }
        
        return max_;
    }
};

File: best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_day = inf
        res = 0
        for i in range(len(prices)):
            
            res = max(res, prices[i] - min_day)
            min_day = min(min_day, prices[i])
        return res
            
            
        